crates/autopilot/src/domain/settlement/auction.rs

```diff
@@ -0,0 +1,27 @@
+//! Auction data related to the specific settlement.
+
+use {
+    crate::domain::{self, competition, eth},
+    std::collections::{HashMap, HashSet},
+};
+
+/// Offchain auction data related to a specific settlement that got settled for
+/// it.
+#[derive(Debug)]
+pub struct Auction {
```

Comment:
I think this should be called sth like `FinalizedAuction` or `ArbitratedAuction` because the regular auction can't have a winner, solution and other things yet.

Reply:
Agree. Initially when implemented it contained only auction data but I added more, without renaming. IMO shouldn't have auction in its name since it contains auction and other data as well, like winning solution or missing order uids etc. Maybe `Competition`

Reply:
This now contains only data related to Auction so kept it as is.

Reply:
I'm still a bit confused how this relates to `domain::auction::Auction`. E.g. why does it not need the orders (to later on check which orders in the solution are allowed to count for surplus)?

I also think the things that are on this struct but not on the other (fee_policies, jit_orders) should probably become part of the core auction as it needs to be decided before the runloop is officially started.

Maybe we can add a note/task that eventually these two structs could potentially be combined (or are at least married in some way).